addons/stock/report/report_stockpicking_operations.xml

```diff
@@ -66,100 +50,156 @@
                                                             (product barcode)
                                                         </div>
                                                     </div>
                                                 </span>
-                                                <t t-set="product_barcode" t-value="ml.product_id.barcode"/>
-                                            </t>
-                                        </td>
-                                    </tr>
+                                            </td>
+                                        </tr>
+                                        <!-- If a move has multiple move lines, we then loop over them and print a breakdown -->
+                                        <t t-foreach="move.move_line_ids" t-as="line" t-if="move_has_multiple_lines">
+                                            <tr>
+                                                <td>
+                                                    <span style="margin-left: 50px;" t-if="line.lot_id or line.lot_name" t-esc="line.lot_id.name"></span>
+                                                </td>
+                                                <td class="text-end">
+                                                    <span t-esc="format_number(line.quantity)">3.00</span>
+                                                    <span t-field="line.product_uom_id" groups="uom.group_uom">units</span>
+                                                    <span t-if="line.move_id.product_packaging_id">
+                                                        (<span t-field="line.product_packaging_qty" t-options='{"widget": "integer"}'/> <span t-field="line.move_id.product_packaging_id.name"/>)
+                                                    </span>
+                                                </td>
+                                                <td t-if="from_col_exists" groups="stock.group_stock_multi_locations">
+                                                    <span t-field="line.location_id.display_name">WH/Stock</span>
+                                                    <t t-if="line.package_id">
+                                                        <span t-field="line.package_id">Package A</span>
+                                                    </t>
+                                                </td>
+                                                <td t-if="to_col_exists" groups="stock.group_stock_multi_locations">
+                                                    <span t-field="line.location_dest_id.display_name">WH/Outgoing</span>
+                                                    <t t-if="line.result_package_id">
+                                                        <span t-field="line.result_package_id">Package A</span>
+                                                    </t>
+                                                </td>
```

Comment:
:crying_cat_face: I just noticed that the existing code is wrong. We can maybe fix it for master since I guess no one has complained about it (i.e. it doesn't need to be fixed in stable). Packages should be visible even if multi-location isn't active (groups="group1,group2" is an or condition), your `from_col_exists` and `to_col_exists` should be adjusted to also check if any of the picking mls have a `package_id` and `result_package_id` respectively. We can assume if there are existing packages, then the setting is probably active (i.e. no need for extra group checks)

```suggestion
                                                <td t-if="from_col_exists" groups="stock.group_stock_multi_locations,stock.group_tracking_lot">
                                                    <span t-field="line.location_id.display_name" groups="stock.group_stock_multi_locations">WH/Stock</span>
                                                    <t t-if="line.package_id">
                                                        <span t-field="line.package_id">Package A</span>
                                                    </t>
                                                </td>
                                                <td t-if="to_col_exists" groups="stock.group_stock_multi_locations,stock.group_tracking_lot">
                                                    <span t-field="line.location_dest_id.display_name" groups="stock.group_stock_multi_locations">WH/Outgoing</span>
                                                    <t t-if="line.result_package_id">
                                                        <span t-field="line.result_package_id">Package A</span>
                                                    </t>
                                                </td>
```

same fix applies to single line move + when to add the "Form"/"To" column headers

Reply:
Hm, now that I see the result, it's not great :skull: now the column will show when packages are enabled + locations are disabled and there are no source packages. Maybe it's better to revert this change unless you can figure out a clean way to have it follow the expected logic:

- if locations active + correct picking type => always show column
- locations not active + packages active + package assigned => show column
- locations not active + packages active + no packages assigned => DON'T show column

I tried a tiny bit to make it work and failed, so good luck, I defer to your decision on it :saluting_face: